connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -724,9 +696,19 @@ private static Object convertToConnect(Schema schema, JsonNode jsonValue) {
                     break;
             }
         }
-        typeConverter = TO_CONNECT_CONVERTERS.get(schemaType);
+
+        final JsonToConnectTypeConverter typeConverter = TO_CONNECT_CONVERTERS.get(schemaType);
         if (typeConverter == null)
-            throw new DataException("Unknown schema type: " + schema.type());
+            throw new DataException("Unknown schema type: " + schemaType);
+
+        if (jsonValue.isNull()) {
+            assert schema != null; // otherwise we would have thrown above because of unknown schema type
```

Comment:
Is this always true? Can't the method be called with `schema` set to null? (The method does check for that condition.)